File: AtCoder/177E.cpp
/* Coprime */

#include <bits/stdc++.h>
using namespace std;

int GCD(vector<int> arr){
    int gcd = arr[0];
    for(auto i = 1; i < (int)arr.size(); i++){
        gcd = __gcd(arr[i],gcd);
    }
    return gcd;
}

void is(vector<int> entry){
    for(auto i = 0; i < (int)entry.size()-1; i++){
        for(auto j = i+1; j < (int)entry.size(); j++){
            if(GCD(entry)==1&&__gcd(entry[i],entry[j])!=1){
                cout << "setwise" << endl;
                return;
            }
            else if(GCD(entry)!=1&&__gcd(entry[i],entry[j])!=1){
                cout << "neither" << endl;
                return;
            }
            else{
                cout << "pairwise" << endl;
                return;
            }
                
        }
    }
}

int main(){
    vector<int> tab1 = {3,4,5}; // pairwise
    vector<int> tab2 = {6,10,15}; // setwise
    vector<int> tab3 = {6,10,16}; //neither
    is(tab1); is(tab2); is(tab3);
}

File: AtCoder/177E.py
# Coprime

import math

def GCD(arr)->int:
	gcd = arr[0]
	for i in range(1,len(arr)):
		gcd = math.gcd(arr[i],gcd)
	return gcd

def is_coprime(entry):
	for i in range(len(entry)-1):
		for j in range(i+1,len(entry)):
			if(GCD(entry)==1 and math.gcd(entry[i],entry[j])!=1):
				print("setwise", end="\n")
				return
			elif(GCD(entry)!=1 and math.gcd(entry[i],entry[j])!=1):
				print("neither", end="\n")
				return
			else:
				print("pairwise", end="\n")
				return
	 
if __name__=='__main__':
	tab1 = [3,4,5] #pairwise
	tab2 = [6,10,15] #setwise
	tab3 = [6,10,16] #neither
	is_coprime(tab1)
	is_coprime(tab2)
	is_coprime(tab3)
